modules/ldsg.js

```diff
@@ -0,0 +1,157 @@
+      if (!member) return int.channel?.send("I could not find that member");
+      try {
+        member.send({ embeds: [em] });
+        return int.channel?.send({ content: "Replied to user:", embeds: [em] });
+      } catch (e) {
+        u.errorHandler(e, int);
+        return int.channel?.send("Failed to message member, they may have me blocked. You will need to reach out to them on your own this time!");
+      }
+    } else if (int.customId == "suggestionManage") {
+      if (!int.channel) return int.reply({ content: "Channel error", ephemeral: true });
+      const post = int.guild.channels.cache.get(int.channel.id);
+      await int.showModal(manageModal);
+      const submitted = await int.awaitModalSubmit({ time: 5 * 60 * 1000, dispose: true }).catch(() => {
+        return null;
+      });
+      if (!submitted) return int.channel?.send("I fell asleep waiting for your input...");
+      await submitted.deferUpdate();
+      const title = submitted.fields.getTextInputValue("renameText");
+      const issue = submitted.fields.getTextInputValue("issueText");
+      const cause = submitted.fields.getTextInputValue("causeText");
+      if (!title && !issue && !cause) return int.followUp({ content: "I need some stuff to change!", ephemeral: true });
+      const fields = int.message.embeds[0].fields;
+      const em = u.embed(int.message.embeds[0]);
+      if (title) {
+        try {
+          const old = post?.name;
+          await post?.setName(title);
+          em.setTitle(title);
+          int.followUp({ content: `> Changed title from "${old}" to "${title}"`, ephemeral: true });
+        } catch (e) {
+          u.errorHandler(e, int);
+          return int.channel.send("Failed to rename forum post");
+        }
+      }
+      if (issue) em.setFields([...fields.filter(f => f.name != "Issue"), { name: "Issue", value: issue }]);
+      if (cause) em.setFields([...fields.filter(f => f.name != "Root Cause"), { name: "Root Cause", value: cause }]);
+      return int.message.edit({ content: int.message.content, embeds: [em], components: replyOption });
+    }
+  } catch (e) { u.noop; }
+}
+
+const Module = new Augur.Module()
+  .addInteraction({
+    name: "ldsg",
+    id: u.sf.commands.slashLdsg,
+    process: async (interaction) => {
+      const subcommand = interaction.options.getSubcommand(true);
+      switch (subcommand) {
+        case "members": return slashLdsgMembers(interaction);
+        case "suggest": return slashLdsgSuggest(interaction);
+      }
+    }
+  })
+  .addEvent("interactionCreate", (int) => {
+    if (!int.inCachedGuild() || !int.isButton() || int.guild.id != u.sf.ldsg) return;
+    if (!u.perms.calc(int.member, ["team", "mgr"])) {
+      return int.reply({ content: "You don't have permissions to interact with this suggestion!", ephemeral: true });
+    }
+    if (int.customId.startsWith("suggestion")) return processCardAction(int);
```

Comment:
order of operation is wrong here. this will effect every interaction

Reply:
oh wow 💀